### PR TITLE
Add breadcrumbs component

### LINKED ISSUE #3965: Breadcrumbs
## 🗒 General
ezgif com-gif-maker

We'd love help building out a breadcrumb component for Backstage! Breadcrumbs take place in the header, above the Title. Our breadcrumbs are expandable and can also show (via dropdown) pages that are on the same hierarchical level.

## 💻 Usage

Breadcrumbs in Header

An example of the logic of our breadcrumbs is as follows:
Logic Examples

## 📐 Specs

Specs can be found by duplicating our [UX kit in Figma](https://www.figma.com/@backstage). Under the 'Headers (w/ Breadcrumbs' page, you'll find breadcrumb specs.

## 🔮 Future

We're playing around with the idea of modifying the header, so breadcrumbs potentially can live outside of the header (above the page title instead). Stay tuned.

### PR DESCRIPTION
**Pull request by tudi2d**
Implemented Breadcrumbs component based on [Figma Design](https://www.figma.com/@backstage). The component is built on top of the MUI Breadcrumbs to enforce the given Design & add the specific Popover behavior for hidden Breadcrumbs.

Closes #3965 

**Light:**
image

**Dark:**
image

**In Header:**
image

**Live example in the TechDocs**
MyGst7X49O

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)

### CODE REVIEW

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4511-maolbu2coelr0fkalaxz63na9gx9q2pp.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/602660ea65f8177010eebf45

File: .changeset/wise-books-turn.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/core': minor
```
Comment (benjdlambert) [inline .changeset/wise-books-turn.md:2]: Whilst we're below <1.0.0 this should be `patch` version :)

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4511-maolbu2coelr0fkalaxz63na9gx9q2pp.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/602660ea65f8177010eebf45

**Comment by benjdlambert** [comment]
Perfect! Nice job! Thanks for this!

So from what I can see the only thing that's left is to hook this component up to the `react-router` history stack?

**Comment by tudi2d** [comment]
> 
> 
> Perfect! Nice job! Thanks for this!
> 
> So from what I can see the only thing that's left is to hook this component up to the `react-router` history stack?
> 
> Not saying we should do that in this PR, as I know this was component only, but just so I know to create a ticket out the back of this one to track the work.

Thanks for reviewing @benjdlambert - appreciate it! 🙂 

This resource could be helpful for an additional ticket: [MUI Breadcrumbs integration with react-router](https://material-ui.com/components/breadcrumbs/#integration-with-react-router). It should work the same way for this component. Additionally it could be nice to replace all instances of the MUI Breadcrumbs component with this custom component to enforce the styling.

**Comment by benjdlambert** [comment]
Perfect - sounds great - I'll take care of it! 👍

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**